### ISSUE TITLE
Terraform emits an error if a template changes but the rendering does not

### ISSUE THREAD

**Issue by jemmyw** [hashicorp/terraform#5735]
If I have a template and I make structural changes to the way it is put together (say, move something to a variable) that results in a change but the same template output, and that template is used for an AWS IAM policy, then terraform will show a change for the policy but the following error will occur when apply is run:

```
Error applying plan:

1 error(s) occurred:

* aws_iam_role.ecs-service-role: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.13
    Resource ID: aws_iam_role.ecs-service-role
    Mismatch reason: attribute mismatch: assume_role_policy
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"assume_role_policy":*terraform.ResourceAttrDiff{Old:"{\n  \"Version\": \"2008-10-17\",\n  \"Statement\": [\n    {\n      \"Action\": \"sts:AssumeRole\",\n      \"Principal\": {\n        \"Service\": \"ecs.amazonaws.com\"\n      },\n      \"Effect\": \"Allow\"\n    }\n  ]\n}\n", New:"${template_file.ecs_assume_role_policy.rendered}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyTainted:false}

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.


Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by jemmyw** [comment]
A second run with the same target will result in "No changes."

**Comment by phinze** [comment]
Thanks for this report @jemmyw - can you provide an example reproducing config to help us look into this?

**Comment by jemmyw** [comment]
Sure this is quite easy to reproduce. Here is the configuration. Run this once and it works fine.

template policies/s3.json
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1451943190000",
            "Effect": "Allow",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::my-bucket"
            ]
        }
    ]
}
```

iam.tf
```
resource "template_file" "s3-policy" {
  template = "${file("policies/s3.json")}"
}

resource "aws_iam_role" "s3-role" {
  name = "s3-role"
}

resource "aws_iam_role_policy" "s3" {
  name = "s3-policy"
  policy = "${template_file.s3-policy.rendered}"
  role = "${aws_iam_role.s3-role.id}"
}
```

Now make a change to the template that makes it ouput the same:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1451943190000",
            "Effect": "Allow",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::my-${"bucket"}"
            ]
        }
    ]
}
```

This will now error on apply.

**Comment by mitchellh** [comment]
This is fixed in 0.7.11. Thanks!
